rust/datafusion/src/physical_plan/hash_join.rs

```diff
@@ -80,6 +80,8 @@ pub struct HashJoinExec {
     schema: SchemaRef,
     /// Build-side
     build_side: Arc<Mutex<Option<JoinLeftData>>>,
+    /// Shares the `RandomState` for the hashing algorithm
+    random_state: RandomState,
```

Comment:
This is needed to keep the random state the same between hashing and probing stage.